### PR TITLE
Add autofix for `PLR1701` (repeated-isinstance-calls)

### PR DESCRIPTION
**Pull request by dhruvmanila**
## Summary

Add auto-fix for `PLR1701` (repeated-isinstance-calls)

## Test Plan

* Updated existing snapshot with `cargo test` & `cargo insta review`
* Added new snapshot for Python version 3.9

resolves: #4768

### CODE REVIEW

File: crates/ruff/src/rules/pylint/rules/repeated_isinstance_calls.rs [hunk 1]
```diff
@@ -38,16 +40,31 @@ use crate::checkers::ast::Checker;
...
+
+fn merged_isinstance_call(
+    obj: &str,
+    types: impl IntoIterator<Item = String>,
+    target_version: PythonVersion,
+) -> String {
+    if target_version >= PythonVersion::Py310 {
+        format!("isinstance({}, {})", obj, types.into_iter().join(" | "))
+    } else {
+        format!("isinstance({}, ({}))", obj, types.into_iter().join(", "))
```
Comment (dhruvmanila) [inline crates/ruff/src/rules/pylint/rules/repeated_isinstance_calls.rs:67]: Should we use the AST to build this or is it fine to format it directly?

Comment (charliermarsh) [reply crates/ruff/src/rules/pylint/rules/repeated_isinstance_calls.rs]: I think either would be reasonable.

File: crates/ruff/src/rules/pylint/rules/repeated_isinstance_calls.rs [hunk 2]
```diff
@@ -74,6 +79,9 @@ pub(crate) fn repeated_isinstance_calls(
         let [obj, types] = &args[..] else {
             continue;
         };
+        if !checker.semantic_model().is_builtin("isinstance") {
+            return;
+        }
```
Comment (dhruvmanila) [inline crates/ruff/src/rules/pylint/rules/repeated_isinstance_calls.rs:84]: @charliermarsh I'm unable to understand why was this moved here 😅

Instead of checking once, this will check for each iteration but I don't think the semantic model changes. Am I missing something here?

Comment (charliermarsh) [reply crates/ruff/src/rules/pylint/rules/repeated_isinstance_calls.rs:84]: It effectively delays the check until we have at least one `isinstance` call. The result should be the same, since we're `return`ing, but we avoid having to do the `is_builtin` check if none of the calls are `isinstance`.

Comment (dhruvmanila) [reply crates/ruff/src/rules/pylint/rules/repeated_isinstance_calls.rs:84]: Oh right, understood. Thanks!
